media/server/gstplayer/source/tasks/generic/HandleBusMessage.cpp

```diff
@@ -161,10 +162,48 @@ void HandleBusMessage::execute() const
         }
         break;
     }
+    case GST_MESSAGE_ERROR:
+    {
+        GError *err = nullptr;
+        gchar *debug = nullptr;
+        m_gstWrapper->gstMessageParseError(m_message, &err, &debug);
+
+        if ((err->domain == GST_STREAM_ERROR) && (allSourcesEos()))
+        {
+            RIALTO_SERVER_LOG_WARN("Got stream error. But all streams are ended, so reporting EOS. Error code %d: %s "
+                                   "(%s).",
+                                   err->code, err->message, debug);
+            if (m_gstPlayerClient)
+            {
+                m_gstPlayerClient->notifyPlaybackState(PlaybackState::END_OF_STREAM);
+            }
+        }
+        else
+        {
+            RIALTO_SERVER_LOG_ERROR("Error %d: %s (%s)", err->code, err->message, debug);
+            m_gstPlayerClient->notifyPlaybackState(PlaybackState::FAILURE);
+        }
+
+        m_glibWrapper->gFree(debug);
+        m_glibWrapper->gErrorFree(err);
+        break;
+    }
     default:
         break;
     }
 
     m_gstWrapper->gstMessageUnref(m_message);
 }
+
+bool HandleBusMessage::allSourcesEos() const
+{
+    for (const auto streamInfo : m_context.streamInfo)
```

Comment:
Are you making copy instead of using reference on purpose?

Reply:
Good spot, not intentional.